test/booksByAuthor.test.js

```diff
@@ -26,6 +26,15 @@ describe('GET /books/author/:authorName', () => {
     res.body.forEach(book => {
       expect(book.author).to.equal(authorName);
     });
+
+  it('should handle invalid author names gracefully', async () => {
+    const invalidAuthorName = '@!#Invalid123';
+    const res = await chai.request(server).get(`/books/author/${invalidAuthorName}`).set('Authorization', `Bearer ${token}`);
+    expect(res).to.have.status(200).or.have.status(400);
+    if (res.status === 200) {
+      expect(res.body).to.be.an('array').that.is.empty;
+    }
+  });
   });
```

Comment:
This `});` should be above the newly created test (wrong indentation here). And the expected status should be 404 only.